pubs/config/spec.py

```diff
@@ -31,6 +31,25 @@
 # the full python stack is printed.
 debug = boolean(default=False)
 
+# If true the citekey is normalized using the 'citekey_format' on adding new publications.
+normalize_citekey = boolean(default=False)
+
+# String specifying how to format the citekey. All strings of 
+# the form '{{substitution:modifier}}' and '{{substitution}}' will 
+# be substituted with their appropriate values. The following
+# substitutions are used:
+#    author_last_name: last name of the first author
+#    year: year of publication
+#    first_word: first word of the title
+# modifiers:
```

Comment:
I think an important modifier that would increase flexibility and automation would be "optional". Include the field if found. Could be used for adding the month, when present, without auto-generation fail on half the entries.